libs/pages/application/src/lib/feature/page-settings-resources-feature/page-settings-resources-feature.tsx

```diff
@@ -0,0 +1,110 @@
+import { convertCpuToVCpu, convertMemory } from '@console/shared/utils'
+import { AppDispatch, RootState } from '@console/store/data'
+import PageSettingsResources from '../../ui/page-settings-resources/page-settings-resources'
+
+export const handleSubmit = (data: FieldValues, application: ApplicationEntity, memorySize: MemorySizeEnum) => {
+  const cloneApplication = Object.assign({}, application)
+  const currentMemory = Number(data['memory'])
+
+  cloneApplication.memory = memorySize === MemorySizeEnum.GB ? currentMemory * 1024 : currentMemory
+  cloneApplication.cpu = convertCpuToVCpu(data['cpu'][0], true)
+  cloneApplication.min_running_instances = data['instances'][0]
+  cloneApplication.max_running_instances = data['instances'][1]
+
+  return cloneApplication
+}
+
+export function PageSettingsResourcesFeature() {
+  const { applicationId = '' } = useParams()
+
+  const [loading, setLoading] = useState(false)
+  const dispatch = useDispatch<AppDispatch>()
+
+  const application = useSelector<RootState, ApplicationEntity | undefined>(
+    (state) => selectApplicationById(state, applicationId),
+    (a, b) =>
+      a?.memory === b?.memory &&
+      a?.cpu === b?.cpu &&
+      a?.min_running_instances === b?.min_running_instances &&
+      a?.max_running_instances === b?.max_running_instances &&
+      JSON.stringify(a?.instances) === JSON.stringify(b?.instances)
+  )
+
+  const methods = useForm({
+    mode: 'onChange',
+    defaultValues: {
+      memory: application?.memory,
+      cpu: [convertCpuToVCpu(application?.cpu)],
+      instances: [application?.min_running_instances || 0, application?.max_running_instances || 1],
+    },
+  })
+
+  const [memorySize, setMemorySize] = useState<MemorySizeEnum>(MemorySizeEnum.MB)
+
+  useEffect(() => {
+    methods.setValue('memory', application?.memory)
+    methods.setValue('cpu', [convertCpuToVCpu(application?.cpu)])
+    methods.setValue('instances', [application?.min_running_instances || 1, application?.max_running_instances || 1])
+  }, [
+    methods,
+    application?.memory,
+    application?.cpu,
+    application?.min_running_instances,
+    application?.max_running_instances,
+  ])
+
+  const handleChangeMemoryUnit = () => {
+    const watchMemory = methods.watch('memory')
+    const newMemorySize = memorySize === MemorySizeEnum.MB ? MemorySizeEnum.GB : MemorySizeEnum.MB
+    setMemorySize(newMemorySize)
+
+    if (newMemorySize === MemorySizeEnum.GB) {
+      const newMemory = convertMemory(Number(watchMemory), MemorySizeEnum.MB)
+      methods.setValue('memory', newMemory)
+    } else {
+      const newMemory = convertMemory(Number(watchMemory), MemorySizeEnum.GB)
+      methods.setValue('memory', newMemory)
+    }
+  }
+
+  const onSubmit = methods.handleSubmit((data) => {
+    if (!application) return
+
+    setLoading(true)
+    const cloneApplication = handleSubmit(data, application, memorySize)
+
+    dispatch(
+      editApplication({
```

Comment:
You could mock the `editApplication` and make sure that it has been called with a given parameter to at the minimum assess that on submit, we dispatch the function with the good values.
I can explain more this way of doing or you can check how I did in advancedSettingsFeature to test the submit in the same way. 

I agree that externalising all the time the function is boring so I was looking for a way to still assert that something happens and happens kind of correctly. I ended up thinking that mocking the action and make sure that we call it with the good payload is convincing enough